### PR TITLE
DCOS-2122 fix dependency path in json-schema (#219) 

### PR DESCRIPTION
**Pull request by tamarrow**

### CODE REVIEW

File: cli/dcoscli/data/marathon-group-schema.json [hunk 1]
```diff
@@ -143,7 +143,7 @@
                 },
                 "dependencies": {
                     "items": {
-                        "$ref": "#/definitions/pathType"
+                        "ref": "#/definitions/pathType"
```
Comment (mgummelt) [inline cli/dcoscli/data/marathon-group-schema.json]: what does dropping the dollar sign do?

Comment (tamarrow) [inline cli/dcoscli/data/marathon-group-schema.json]: makes the path relative

Comment (mgummelt) [inline cli/dcoscli/data/marathon-group-schema.json]: I don't see that in the jsonschema docs.  Can you point me to it?

File: cli/tests/data/marathon/apps/zero_instance_sleep.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "cpus": 0.1,
   "mem": 16,
   "instances": 0,
+  "dependencies": ["/product/database", "../backend"],
```
Comment (mgummelt) [inline cli/tests/data/marathon/apps/zero_instance_sleep.json]: Do we ever deploy this app?  It won't deploy with missing deps, right?

File: cli/dcoscli/data/marathon-group-schema.json (hunk 1 above)
Comment (tamarrow) [inline cli/dcoscli/data/marathon-group-schema.json]: I saw it on stackoverflow. I dont remember where, I'll look for it

File: cli/tests/data/marathon/apps/zero_instance_sleep.json (hunk 2 above)
Comment (tamarrow) [inline cli/tests/data/marathon/apps/zero_instance_sleep.json]: no

Comment (mgummelt) [inline cli/tests/data/marathon/apps/zero_instance_sleep.json]: which part?

Comment (tamarrow) [inline cli/tests/data/marathon/apps/zero_instance_sleep.json]: no, yes respectively

File: cli/dcoscli/data/marathon-group-schema.json [hunk 3]
```diff
@@ -143,7 +137,8 @@
                 },
                 "dependencies": {
                     "items": {
-                        "$ref": "#/definitions/pathType"
+                        "pattern": "^(\\/?((\\.{2})|([a-z0-9\\-]*))($|\\/))*$",
+                        "type": "string"
```
Comment (tamarrow) [inline cli/dcoscli/data/marathon-group-schema.json]: I just moved the pathType info here, since it's only used once, and this makes things simpler.

Comment (mgummelt) [inline cli/dcoscli/data/marathon-group-schema.json]: lol, OK

**Comment by tamarrow** [comment]
@mgummelt tests passed. okay to merge?

**Comment by mgummelt** [comment]
yep

**Pull request merged**